docs/specification/draft/basic/streams.mdx

```diff
@@ -0,0 +1,240 @@
+    Client->>+Server: Request (e.g., tools/call)
+    Server-->>Client: notifications/stream/create
+
+    Note over Client,Server: Stream Operation
+    loop Message Exchange
+        Server-->>Client: Messages on stream
+    end
+
+    Note over Client,Server: Stream Termination
+    Server-->>-Client: notifications/stream/end
+```
+
+### Creation
+
+When a client sends a request that may produce a long-running response, the server can create a stream by sending a `notifications/stream/create` notification with a unique stream ID and resume interval parameters.
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "notifications/stream/create",
+  "params": {
+    "stream": {
+      "streamId": "550e8400-e29b-41d4-a716-446655440000",
+      "resumeInterval": {
+        "min": 5,
+        "max": 3600
+      }
+    }
+  }
+}
+```
+
+The `resumeInterval` specifies:
+- `min`: Minimum seconds a client should wait before resuming the stream (prevents excessive reconnections)
+- `max`: Maximum seconds a client may wait before the server considers the stream abandoned
+
+### Disconnection and Resumption
+
+Either the client or server may disconnect at any time. After a disconnection, the client can resume the stream by sending a `stream/resume` request with the stream ID:
+
+```mermaid
+sequenceDiagram
+    participant Client
+    participant Server
+
+    Client->>+Server: Request (e.g., tools/call)
+    Server-->>Client: notifications/stream/create
+    Server-->>Client: Initial messages
+
+    Note over Client,Server: Disconnection occurs
+
+    Client->>Server: stream/resume
+    Server-->>Client: Unsent messages
+    Server-->>-Client: notifications/stream/end
+```
+
+Example `stream/resume` request:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "stream/resume",
+  "params": {
+    "streamId": "550e8400-e29b-41d4-a716-446655440000"
+  }
+}
+```
+
+When a client sends a `stream/resume` request, the server should:
+1. Send all unsent messages for the stream to the client
+2. If the stream has ended, send a `notifications/stream/end` notification after sending all other unsent messages
+3. If the stream ID is invalid (e.g., expired), respond with an error
+
+
+#### Streamable HTTP Considerations
+
+When using the Streamable HTTP transport, streams are implemented using the Server-Sent Events (SSE) protocol. When served over HTTP, a server cannot definitively when any given message is delivered. Clients use the `Last-Event-ID` HTTP header to resume their SSE connection to Streamable HTTP servers.
+
+In order to deliver durable message guarantees, servers may maintain a buffer of recently sent messages (a two minute window aligns with common TCP idle timeout configurations), and use a scheme to correlate message IDs across streams, such as by using a strictly monotonic counter. When a client reconnects after a disconnection, the `Last-Event-ID` header can be used to derive the point from which its view of streams will resume and poll from.
```

Comment:
Note for reviewers: see also https://github.com/modelcontextprotocol/modelcontextprotocol/discussions/543#discussioncomment-13265363.